File: spring-backend/src/main/java/com/recover/project/dto/project/CreateProject.java
package com.recover.project.dto.project;

import java.time.LocalDate;
import java.util.Set;

import com.google.auto.value.AutoValue.Builder;
import com.recover.project.dto.user.ProjectRoleRequest;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.Data;

@Data
@Builder
public class CreateProject {
   @NotBlank(message = "Project name is required")
   private String projectName;
   
   @NotNull(message = "Start date is required")
   private LocalDate startDate;

   @NotNull(message = "Loss date is required") 
   private LocalDate lossDate;

   @NotNull
   private LocalDate receivedDate;

   private LocalDate policyStart;

   private LocalDate policyExpiration;

   private String claimNumber;

   private String yearBuilt;

   private String catReference;

   private String carrier;

   @NotNull(message = "Client name is required") 
   private String clientName;

   @NotNull(message = "Client email is required") 
   private String clientEmail; 

   @NotNull(message = "Client phone is required") 
   private String clientPhone;

   private String streetAddress;

   private String city;

   private String state;

   private String zipcode;

   private String stage;

   private String projectType;

   private String lossType;

   private String scope;

   private Set<ProjectRoleRequest> assignedUsers;
   
}

File: frontend/src/types/createProject.ts
import { Dispatch, SetStateAction } from "react";

export interface CreateProject {
    projectName: string;
    //startDate: string;
    lossDate: string;
    clientName: string;
    clientEmail: string;
    clientPhone: string;
    streetAddress: string;
    city: string;
    state: string;
    zipcode: string;
    stage: string;
    projectType: string;
    carrier: string;
    //assignedUser: string[]; //??? maybe add later
    //recievedDate: string; //Dave has this autosetting in backend
    catReference?: string;
    lossType: string;
    scope: string;
    claimNumber: string;
    policyExpiration: string;
    policyStart: string;
    yearBuilt: number;
    //officeId: string; //TODO ADD LATER? number??
}

// export interface ProjectRoleRequest {
//     userId: string;
//     roleType: string;
//     projectId: string;
// }

export interface CreateProjectError {
    message: string;
    field?: string;
}

export interface ProjectStageArray {
    label: string;
    key: string;
}
export interface LossTypeArray {
    label: string;
    key: string;
}

export interface StepOneProps {
    firstName: string;
    lastName: string;
    formData: CreateProject;
    errors: CreateProjectError | null;
    handleKeyDown: (e: React.KeyboardEvent<HTMLInputElement>) => void;
    handleInputChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
}

export interface StepTwoProps {
    projectType: string[];
    formData: CreateProject;
    errors: CreateProjectError | null;
    handleInputChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
    handleKeyDown: (e: React.KeyboardEvent<HTMLInputElement>) => void;
    handleProjectTypeChange: (projectType: string) => void;
}

export interface StepThreeProps {
    scope: string[];
    lossType: string[];
    formData: CreateProject;
    errors: CreateProjectError | null;
    handleInputChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
    handleKeyDown: (e: React.KeyboardEvent<HTMLInputElement>) => void;
    setErrors: Dispatch<SetStateAction<CreateProjectError | null>>;
}

export interface StepFourProps {
    formData: CreateProject
    errors: CreateProjectError | null;
    handleInputChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
    handleKeyDown: (e: React.KeyboardEvent<HTMLInputElement>) => void;
}

export interface StepFiveProps {
    formData: CreateProject
    projStage: string[];
    handleStageChange: (projectStage: string) => void;
}